### ISSUE TITLE
[HOLD for payment 2023-10-23] [HOLD for payment 2023-10-23] [HOLD for payment 2023-10-20] [HOLD for payment 2023-10-20] [$500] The "Hmm... it's not here" message does not appear when opening the assignee modal of a canceled task using the URL

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#28026]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Start a chat.
2. Navigate to the "Assign task" section.
3. Enter a title and create the task.
4. After task creation, click on the assignee.
5. Copy the URL of the assignee modal and send it.
6. Proceed to cancel the task.
7. After canceling, click on the sent link
## Expected Result: 
Upon opening the assignee page of the canceled task using the URL, the "Hmm... it's not here" message should be displayed for the assignee
## Actual Result: 
The "Hmm... it's not here" message does not appear when opening the assignee modal of a canceled task using the URL

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.73-0
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/8a14b504-f6a0-4386-a3eb-bdbda9847053

https://github.com/Expensify/App/assets/43996225/166427df-ad40-46d6-8261-e0f5128ac84b

**Expensify/Expensify Issue URL:**
**Issue reported by:** @ayazhussain79
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1695054791232689

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0181644eb19fd0aac9</li>
        <li>Upwork Job ID: 1705246939905851392</li>
        <li>Last Price Increase: 2023-09-29</li>
<li>Automatic offers: </li>
<ul>
<li>ahmedGaber93 | Contributor | 27047586</li>
</ul></li>
    </ul>
</details>

**Comment by ahmedGaber93** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The "Hmm... it's not here" message does not appear when opening the assignee modal of a canceled task using the URL
### What is the root cause of that problem?
The root cause is when a task is deleted, the `parentReportAction` is updated to have a isDeletedParentAction deleted flag
but here when we check if task `isOpen` we use the `task.report` instead `report`
https://github.com/Expensify/App/blob/496943389debfb0bc0dc453da81950e28ceebad2/src/pages/tasks/TaskAssigneeSelectorModal.js#L201-L203

isOpenTaskReport 
https://github.com/Expensify/App/blob/496943389debfb0bc0dc453da81950e28ceebad2/src/libs/ReportUtils.js#L220-L222

isOpenTaskReport use isCanceledTaskReport which failed in task.report
https://github.com/Expensify/App/blob/496943389debfb0bc0dc453da81950e28ceebad2/src/libs/ReportUtils.js#L201-L211


### What changes do you think we should make in order to solve the problem?
We need to use `report` instead of `task.report` when we check `isOpen`. 
we can also use it in `canModifyTask` and `isTaskReport` (optional)
```javascript
// we use the same page for both new task asignee and edit it,
// so we need to check it because in new task report will be null 
// we can also use Boolean(report)
const isOpen = report && ReportUtils.isOpenTaskReport(report);
```

### What alternative solutions did you explore? (Optional)
N/A

**Comment by miljakljajic** [comment]
Triaged and reproduced: 
<img width="1437" alt="image" src="https://github.com/Expensify/App/assets/28715117/b2b96df5-d968-4f23-9e30-6eb2836c805a">

**Comment by miljakljajic** [comment]
@sophiepintoraetz do we both need to be assigned to this? If not, feel free to unassign yourself.

**Comment by sophiepintoraetz** [comment]
@miljakljajic - no, we don't 😂 I can take this seeing as I was assigned first (thank you for replicating, as an FYI, if Applause has applied the `External` label, this implies they are able to reproduce the bug and so we don't have to)

**Comment by miljakljajic** [comment]
Oh brilliant I had no idea, thank you!

**Comment by miljakljajic** [comment]
Appreciate you grabbing it also <3

**Comment by sophiepintoraetz** [comment]
Waiting for @robertKozik to review the proposals!

**Comment by sophiepintoraetz** [comment]
Still waiting on @robertKozik here!

**Comment by sophiepintoraetz** [comment]
@isabelastisser - just need a hand babysitting this issue while I'm OOO till 15 Oct
- Waiting on @robertKozik to review the proposals
- I've also [bumped in slack](https://expensify.slack.com/archives/C02NK2DQWUX/p1696480725547259)
- From there, any payments are due (the reporter and potential contributor)

**Comment by robertKozik** [comment]
Hi all! I have a problem with reproducting this issue on latest main, could you check on your end as well whether this issue still exists? Attaching a video from my attempt


https://github.com/Expensify/App/assets/61146594/e1e2de5a-7d61-44ec-80c8-2593a049f09b

**Comment by robertKozik** [comment]
Thanks for pointing it out!

**Comment by robertKozik** [comment]
Yeah, I'm able to reproduce it. Thanks!

I've checked both of your proposals and as I see both of you correctly pinpointed the root cause of the problem. I like more the proposal from @ahmedGaber93, as it solves the issue and mimics the approach used on task title and description modals. 

🎀 👀 🎀 C+ reviewed

**Comment by ahmedGaber93** [comment]
@robertKozik PR https://github.com/Expensify/App/pull/29029 is ready for review.

**Comment by sophiepintoraetz** [comment]
Summary of payments:
- Reporter: @ayazhussain79  - $50
- Contributor: @ahmedGaber93 $500 (plus $250 urgency bonus)
- C+: N/A


Eligible for urgency bonus? Y

**Comment by ayazhussain79** [comment]
@sophiepintoraetz offer accepted, Thank you
